### PR TITLE
switch user.current -> homedir.dir

### PR DESCRIPTION
**Pull request by jsloyer**
The call `user.Current` depends on cgo to compile but when you cross compile on OSX this is not available.  Per this [github issue](https://github.com/ncw/rclone/issues/15) there is a library that can get the home dir without using cgo.  The library is https://github.com/mitchellh/go-homedir and its licensed under MIT.

### CODE REVIEW

**Comment by renier** [comment]
@jsloyer I'm on OSX and it works fine. Don't understand the issue.

**Comment by jsloyer** [comment]
So this is an issue if you cross compile the softlayer go library

**Comment by renier** [comment]
@jsloyer You said cross compile to OSX, so that's the type of binary I'm running, and not experiencing the issue.

Are you using the released binaries? Or are you building it on your own?

**Comment by renier** [comment]
@jsloyer Basically, tell me specifically how to reproduce the problem you saw.

**Comment by jsloyer** [comment]
So I'm running non OSX and if cross compile a binary that includes this library for Linux is when the issue arises

**Comment by renier** [comment]
@jsloyer This is what I did to attempt reproduction:

1. Created a small snippet of code that uses _user.Current()_:
```go
package main

import (
	"fmt"
	"log"
	"os/user"
)

func main() {
	u, err := user.Current()
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(u.HomeDir)
}
```
1. Build it on OSX, with linux as the xcompile target: `env GOOS="linux" GOARCH="amd64" go build`
1. Copy the binary to a Linux system and I run it.

The output is what I expected.

**Comment by jsloyer** [comment]
This docker image reproduces it, https://hub.docker.com/r/jsloyer/user-current-error/

**Comment by renier** [comment]
@jsloyer Can I see the Dockerfile and how you build it?

**Comment by renier** [comment]
@jsloyer I see the issue. Happens executing the cross-compiled binary in a linux container only, but does not happen in a full-blown VM. I which I knew what was missing in the container, so that we could just try adding it.

I see two solutions:

1. You build the binary within a linux container. Avoids cross-compile. [This article describes that approach](https://developer.atlassian.com/blog/2015/07/osx-static-golang-binaries-with-docker/).
2. We retool the logic in softlayer go to, **only if user.Current() fails**, try looking at `$HOME` or `$USERPROFILE` as done [here](https://github.com/mesosphere/mesos-dns/commit/4b99d6cc16c2be170525cd572f6564b673ac90d0).

I don't want to add a dependency just to solve this edge case. We have kept the library largely dependency-free and want to keep it that way.

**Comment by jsloyer** [comment]
I honestly like #2 better.  I update the PR...

File: session/session.go [hunk 1]
```diff
@@ -138,12 +138,13 @@ func New(args ...interface{}) *Session {
 	envFallback("SOFTLAYER_TIMEOUT", &values[keys["timeout"]])
 
 	// Read ~/.softlayer for configuration
-	u, err := user.Current()
+	dir, err := homedir.Dir()
+	fmt.Println(err)
```
Comment (seh) [inline session/session.go]: - [ ] Did you intend to leave that here?  
  At this point, we don't know whether `err` is nil or not.

**Comment by seh** [comment]
Don't replicate [this mistake](https://github.com/mesosphere/mesos-dns/commit/4b99d6cc16c2be170525cd572f6564b673ac90d0#commitcomment-20214672), though.

**Comment by renier** [comment]
@jsloyer Blast from the past, but.. It was never clear whether you had tried building with `CGO_ENABLED=0`.

**Pull request merged**